### PR TITLE
[release/7.0] Move to an RTM SDK

### PR DESCRIPTION
**Pull request by dougbu**

### CODE REVIEW

**Review by wtgodbe — APPROVED** [review]

File: global.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
   "sdk": {
-    "version": "7.0.100-rc.1.22431.12"
+    "version": "7.0.100-rtm.22477.16"
```
Comment (wtgodbe) [inline global.json:3]:
```suggestion
    "version": "7.0.100-rtm.22478.12"
```

File: global.json [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
   },
   "tools": {
-    "dotnet": "7.0.100-rc.1.22431.12",
+    "dotnet": "7.0.100-rtm.22477.16",
```
Comment (wtgodbe) [inline global.json:6]:
```suggestion
    "dotnet": "7.0.100-rtm.22478.12",
```

**Review by wtgodbe — CHANGES_REQUESTED** [review]
I guess `7.0.100-rtm.22477.16` is from the internal branch, `7.0.100-rtm.22478.12` is the latest I could find from a build of `release/7.0.1xx`

**Review by wtgodbe — APPROVED** [review]

**Comment by dougbu** [comment]
> I guess 7.0.100-rtm.22477.16 is from the internal branch

It's the version https://aka.ms/dotnet/7.0.1xx/daily/productCommit-win-x64.txt shows. I asked @marcpopMSFT in FR whether we should prefer that or the version from https://github.com/dotnet/installer. Also asking FR people how previous problems like this have been resolved.

**Comment by wtgodbe** [comment]
> It's the version https://aka.ms/dotnet/7.0.1xx/daily/productCommit-win-x64.txt shows. I asked @marcpopMSFT in FR whether we should prefer that or the version from https://github.com/dotnet/installer. Also asking FR people how previous problems like this have been resolved.

That looks like a bug in the table, here's the build that produced it (note it's of internal/release/7.0.1xx): https://dev.azure.com/dnceng/internal/_build/results?buildId=2005514&view=results

**Comment by dougbu** [comment]
https://aka.ms/dotnet/7.0.1xx/daily/productCommit-win-x64.txt better not refer to internal builds because that's a public file ☹️ In other words, I'm _hoping_ there's an issue somewhere other than in pushing that version of the .txt file.

**Comment by dougbu** [comment]
Think I figured it out. I took the version from the SDK line in productCommit-win-x64.txt instead of the installer line.

Note we normally build and make daily builds public from the internal/&hellip; branches. It's only internal & secure builds that won't touch the `dotnet7` or `dotnet7-transport` feeds. Right @mmitche❔

As far as I can see, nothing is wrong w/ either dotnet/installer or the productCommit-win-x64.txt file. @marcpopMSFT you can stand down 😁

**Comment by mmitche** [comment]
> Think I figured it out. I took the version from the SDK line in productCommit-win-x64.txt instead of the installer line.
> 
> Note we normally build and make daily builds public from the internal/… branches. It's only internal & secure builds that won't touch the `dotnet7` or `dotnet7-transport` feeds. Right @mmitche❔
> 
> As far as I can see, nothing is wrong w/ either dotnet/installer or the productCommit-win-x64.txt file. @marcpopMSFT you can stand down 😁

The internal/* branches are never public. Only the release/* branches are.

**Comment by dougbu** [comment]
Most template tests failing w/

<details>
<summary>Failure to uninstall</summary>

``` text
2022-09-28T23:00:37.0513990Z   Error Message:
2022-09-28T23:00:37.0514160Z    System.InvalidOperationException : Failed to uninstall previous templates. The template 'web' could still be found.
2022-09-28T23:00:37.0514210Z   Stack Trace:
2022-09-28T23:00:37.0514330Z      at Templates.Test.Helpers.TemplatePackageInstaller.VerifyCannotFindTemplateAsync(ITestOutputHelper output, String templateName) in /_/src/ProjectTemplates/Shared/TemplatePackageInstaller.cs:line 144
```
</details>

<details>
<summary>Failure to install</summary>

``` text
2022-09-28T23:00:37.0518130Z  | [0.029s] Templates.Mvc.Test.RazorPagesTemplateTest Information: ==> /private/tmp/helix/working/B0BD097A/p/dotnet-cli/dotnet new "web" --debug:disable-sdk-templates --debug:custom-hive "/private/tmp/helix/working/B0BD097A/w/AA620921/e/Hives/.templateEngine" [/private/tmp/helix/working/B0BD097A/w/AA620921/e/]
2022-09-28T23:00:37.0518200Z  | [0.355s] Templates.Mvc.Test.RazorPagesTemplateTest Information: [ERROR] No templates or subcommands found matching: 'web'.
2022-09-28T23:00:37.0518240Z  | [0.361s] Templates.Mvc.Test.RazorPagesTemplateTest Information: [ERROR] 
2022-09-28T23:00:37.0518280Z  | [0.362s] Templates.Mvc.Test.RazorPagesTemplateTest Information: [ERROR] To list installed templates similar to 'web', run:
2022-09-28T23:00:37.0518330Z  | [0.363s] Templates.Mvc.Test.RazorPagesTemplateTest Information: [ERROR]    dotnet new list web
2022-09-28T23:00:37.0518510Z  | [0.363s] Templates.Mvc.Test.RazorPagesTemplateTest Information: [ERROR] To search for the templates on NuGet.org, run:
2022-09-28T23:00:37.0518550Z  | [0.363s] Templates.Mvc.Test.RazorPagesTemplateTest Information: [ERROR]    dotnet new search web
2022-09-28T23:00:37.0518590Z  | [0.363s] Templates.Mvc.Test.RazorPagesTemplateTest Information: [ERROR] 
2022-09-28T23:00:37.0518620Z  | [0.365s] Templates.Mvc.Test.RazorPagesTemplateTest Information: [ERROR] 
2022-09-28T23:00:37.0518660Z  | [0.365s] Templates.Mvc.Test.RazorPagesTemplateTest Information: [ERROR] For details on the exit code, refer to https://aka.ms/templating-exit-codes#103
```
</details>

I'll check for differences in the relevant files between release/7.0 and main.

**Comment by dougbu** [comment]
Will retry after #44246 is in; that may help here.

**Comment by dougbu** [comment]
@wtgodbe and @BrennanConroy thanks very much for your help w/ this❕

**Pull request merged**